### PR TITLE
overlord/snapstate: misc fixes/tweaks/cleanups

### PR DESCRIPTION
**Pull request by chipaca**
overlord/snapstate: introduce `fakeOps` to have methods on `[]fakeOp` for easier-to-read tests. Rename `(*SnapState).findIndex` to `(*SnapState).LastIndex`.
overlord/snapstate/backend: fail data copy on unexpected copy error.

### CODE REVIEW

**Comment by chipaca** [comment]
note the findIndex -> LastIndex is also a subtle (but soon to be not-so-subtle) functional change as well

File: overlord/snapstate/snapmgr_test.go [hunk 1]
```diff
@@ -140,6 +140,15 @@ func verifyInstallUpdateTasks(c *C, curActive bool, ts *state.TaskSet, st *state
 	return n
 }
 
+func (s *snapmgrTestSuite) TestLastIndexFindsLast(c *C) {
+	snapst := &snapstate.SnapState{Sequence: []*snap.SideInfo{
+		{Revision: snap.R(7)},
+		{Revision: snap.R(11)},
```
Comment (mvo5) [inline overlord/snapstate/snapmgr_test.go]: Aha, this is legal now? multiple revisions in the sequence?

File: overlord/snapstate/backend_test.go [hunk 2]
```diff
@@ -45,6 +45,37 @@ type fakeOp struct {
...
+func (ops fakeOps) Ops() []string {
+	opsOps := make([]string, len(ops))
+	for i, op := range ops {
+		opsOps[i] = op.op
+	}
+
+	return opsOps
+}
+
+func (ops fakeOps) Count(op string) int {
```
Comment (mvo5) [inline overlord/snapstate/backend_test.go]: This (and First()) are nice, however not used in this PR. I assume this will be used in furture PRs?

**Comment by mvo5** [comment]
Looks very nice, thank you.

File: overlord/snapstate/snapmgr_test.go (hunk 1 above)
Comment (chipaca) [inline overlord/snapstate/snapmgr_test.go]: that's what a sequence *is*?

File: overlord/snapstate/backend_test.go (hunk 2 above)
Comment (chipaca) [inline overlord/snapstate/backend_test.go]: yep! used in the aba branch for one (this is part of my whittling the aba branch)

File: overlord/snapstate/snapmgr_test.go (hunk 1 above)
Comment (niemeyer) [inline overlord/snapstate/snapmgr_test.go]: The question is probably about the fact we have multiple 11 entries here. I also assumed earlier that this state would only be reached by mistake.

**Pull request merged**